commit 929c17c86dc045e8d599468ca9b7b71bb19ac2cf
Author: Volker Krause <vkrause@kde.org>
Date:   Mon Oct 28 17:03:59 2024 +0100

    Fix exporting the Android integration singleton to QML
    
    This fixes accessing date/time controls on Android, which broke in 1.5.0.
    
    BUG: 495468

diff --git a/src/dateandtime/CMakeLists.txt b/src/dateandtime/CMakeLists.txt
index 7c6d1b9..8803367 100644
--- a/src/dateandtime/CMakeLists.txt
+++ b/src/dateandtime/CMakeLists.txt
@@ -4,6 +4,7 @@
 if (ANDROID)
     add_library(KF6KirigamiDateAndTime)
     add_library(KF6::KirigamiDateAndTime ALIAS KF6KirigamiDateAndTime)
+    qt_extract_metatypes(KF6KirigamiDateAndTime)
     target_sources(KF6KirigamiDateAndTime PRIVATE
         androidintegration.cpp
     )
diff --git a/src/dateandtime/types.h b/src/dateandtime/types.h
index ffc3382..663aafe 100644
--- a/src/dateandtime/types.h
+++ b/src/dateandtime/types.h
@@ -2,14 +2,24 @@
 // SPDX-License-Identifier: LGPL-2.0-or-later
 
 #include <qqmlregistration.h>
+#include <QQmlEngine>
 
 #ifdef Q_OS_ANDROID
 #include "androidintegration.h"
 
-struct AndroidIntegrationForeign : public QObject
+struct AndroidIntegrationForeign
 {
-    Q_OBJECT
+    Q_GADGET
+    QML_FOREIGN(KirigamiAddonsDateAndTime::AndroidIntegration)
     QML_NAMED_ELEMENT(AndroidIntegration)
     QML_SINGLETON
+
+public:
+    static KirigamiAddonsDateAndTime::AndroidIntegration *create(QQmlEngine *, QJSEngine *)
+    {
+        auto instance = &KirigamiAddonsDateAndTime::AndroidIntegration::instance();
+        QJSEngine::setObjectOwnership(instance, QJSEngine::CppOwnership);
+        return instance;
+    }
 };
 #endif
diff --git a/src/formcard/CMakeLists.txt b/src/formcard/CMakeLists.txt
index 218721f..af2ebe9 100644
--- a/src/formcard/CMakeLists.txt
+++ b/src/formcard/CMakeLists.txt
@@ -11,6 +11,7 @@ ecm_add_qml_module(formcardplugin
         org.kde.coreaddons
         org.kde.kirigami
         org.kde.kirigamiaddons.components
+        org.kde.kirigamiaddons.dateandtime
         org.kde.kirigamiaddons.delegates
 )
 
